web/html/app_middleware.go

```diff
@@ -0,0 +1,73 @@
+package html
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/madappgang/identifo/model"
+	"github.com/urfave/negroni"
+)
+
+const (
+	// FormKeyAppID is a form key to keep application ID.
+	FormKeyAppID = "appId"
+)
+
+// AppID gets app id from the request body.
+func (ar *Router) AppID() negroni.HandlerFunc {
```

Comment:
I suppose we already have auth middleware implemented before?

Reply:
It isn't auth middleware, this middleware check's appID param and puts app in the context.

Reply:
we already have app middleware in API router. Looks like duplicating of the same logic

Reply:
Yes, you're right. I'll reuse this logic

Reply:
But it still gonna be two middlewares, because they have different sources and different ways of handling errors.

Reply:
Yes, you need to extract common logic and reuse it in both places

Reply:
Roger that!